File: Data Structures & Algorithms/pascals-triangle-ii/submission-0.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<vector<int>> res;
        res.push_back({1});

        for (int i = 1; i <= rowIndex; i++) {  

            vector<int> temp;

            for (int j = 0; j < i + 1; j++) {  

                if (j == 0 || j == i)           
                    temp.push_back(1);
                else
                    temp.push_back(res[i-1][j-1] + res[i-1][j]);  
            }

            res.push_back(temp);
        }

        return res[rowIndex];
    }
};

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []
        q = [root]
        indx = 0
        res = []

        while indx < len(q):
            temp = len(q) - indx
            res.append(q[-1].val)
            for j in range(temp):
                if q[indx].left:
                    q.append(q[indx].left)
                if q[indx].right:
                    q.append(q[indx].right)
                indx += 1
        return res